src/Tasks/AssemblyDependency/Resolver.cs

```diff
@@ -95,6 +95,11 @@ protected Resolver(string searchPathElement, GetAssemblyName getAssemblyName, Fi
             out string foundPath,
             out bool userRequestedSpecificFile);
 
+        /// <summary>
+        /// The parent assembly that was used for the SearchPath.
+        /// </summary>
+        internal string ParentAssembly { get; set; }
```

Comment:
super-nit: It would be nicer if this was declared as a read-only property. For example, `parentAssembly` could be added as a constructor parameter on `DirectoryResolver` (because it's not used by any other resolver) and this property would become virtual, returning null here and overriden in `DirectoryResolver` to return what went into the constructor. Totally optional, looks good as is.

Reply:
Done

Reply:
I realised that `parentAssembly` is needed only for `DirectoryResolver` and removed it from `Resolver`.